### PR TITLE
Announce Carbon's participation in GSoC 2023

### PR DESCRIPTION
**Pull request by josh11b**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,11 @@ SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
   <a href="#join-us">Join us</a>
 </p>
 
+**Carbon is a
```
Comment (jonmeow) [inline README.md]: How prominently do we want to advertise Google programs? Maybe move this to #join-us?

Comment (jonmeow) [reply README.md]: Just for comparison, I'd noticed Swift has https://www.swift.org/gsoc2023/, but I'm not sure they have links (https://github.com/apple/swift-org-website/pull/225 didn't add any). It's mostly discussed on their forums.

Comment (josh11b) [reply README.md]: Done.

**Review by wolffg — APPROVED** [review]
LGTM.

**Review by jonmeow — APPROVED** [review]

**Pull request merged**